### DISCUSSION TITLE
How to connect to workspace using dbx and service principle?

### DISCUSSION THREAD

**Discussion by ssabzevari-antuit** [databrickslabs/dbx#702] [General]
I am using GitHub actions and dbx to do unit test and integration test in our Databricks workspace. Currently I am using my personal access token and everything is running fine. However, I want to replace it with a service principle. So far I haven't had a luck with that. My question is:

- Is dbx able to connect using service principles?
- Is there any document or tutorial?

**Comment by ronaldsholt** [comment]
Yes. In Azure, you can use the tenant and client id with your SP's secret [+ add them as variables in your pipeline] in your az-pipelines.yml like this:

```bash
    - bash: |
          echo "##vso[task.setvariable variable=DATABRICKS_TOKEN;issecret=true]$(python path/to/your/script/to/get_token.py)"
      env:
        SP_TENANT_ID: $(SP_TENANT_ID)
        SP_CLIENT_ID: $(SP_CLIENT_ID)
        SP_CLIENT_SECRET: $(SP_CLIENT_SECRET)
      displayName: "Get service principal access token" 
```
